x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/conversations/index.ts

```diff
@@ -128,7 +129,7 @@ export class AIAssistantConversationsDataClient extends AIAssistantDataClient {
       conversationIndex: this.indexTemplateAndPattern.alias,
       conversationUpdateProps,
       isPatch,
-      user: authenticatedUser,
+      user: authenticatedUser ?? this.options.currentUser ?? undefined,
```

Comment:
Do we want to add the same authUser check here for `updateConversation()` as you did above for `createConversation()`?

```ts
if (!this.options.currentUser) {
  throw new Error('AIAssistantConversationsDataClient currentUser is not defined.');
}
```